src/System.Management.Automation/security/Authenticode.cs

```diff
@@ -437,89 +443,83 @@ private static Signature GetSignatureFromWinVerifyTrust(string fileName, string
         }
 
         [ArchitectureSensitive]
-        private static DWORD GetWinTrustData(string fileName, string fileContent,
-                                            out NativeMethods.WINTRUST_DATA wtData)
+        private static uint GetWinTrustData(string fileName, string fileContent,
+                                            out WinTrustMethods.WINTRUST_DATA wtData)
         {
-            DWORD dwResult = Win32Errors.E_FAIL;
-            IntPtr WINTRUST_ACTION_GENERIC_VERIFY_V2 = IntPtr.Zero;
-            IntPtr wtdBuffer = IntPtr.Zero;
-
-            Guid actionVerify =
-                new Guid("00AAC56B-CD44-11d0-8CC2-00C04FC295EE");
-
-            try
+            wtData = new()
             {
-                WINTRUST_ACTION_GENERIC_VERIFY_V2 =
-                    Marshal.AllocCoTaskMem(Marshal.SizeOf(actionVerify));
-                Marshal.StructureToPtr(actionVerify,
-                                       WINTRUST_ACTION_GENERIC_VERIFY_V2,
-                                       false);
-
-                NativeMethods.WINTRUST_DATA wtd;
-
-                if (fileContent == null)
-                {
-                    NativeMethods.WINTRUST_FILE_INFO wfi = NativeMethods.InitWintrustFileInfoStruct(fileName);
-                    wtd = NativeMethods.InitWintrustDataStructFromFile(wfi);
-                }
-                else
+                cbStruct = (uint)Marshal.SizeOf<WinTrustMethods.WINTRUST_DATA>(),
+                dwUIChoice = WinTrustUIChoice.WTD_UI_NONE,
+                dwStateAction = WinTrustAction.WTD_STATEACTION_VERIFY,
+            };
+
+            byte[] contentBytes = fileContent == null
+                ? Array.Empty<byte>()
+                : System.Text.Encoding.Unicode.GetBytes(fileContent);
+            unsafe
+            {
+                fixed (char* fileNamePtr = fileName)
+                fixed (byte* contentPtr = contentBytes)
```

Comment:
`contentBytes` is not used until further below, so does not need to be declared here, and should be moved down where it is used.

Reply:
Sure, the only reason why I did this was to avoid even more indentation but will fix it separately when it's needed rather than all the time.